src/Components/SolutionExplorer.fs

```diff
@@ -195,4 +269,60 @@ module SolutionExplorer =
         window.registerTreeDataProvider("ionide.projectExplorer", provider )
         |> ignore
 
+        let wsProvider =
+            let viewLoading path =
+                "<b>Status:</b> loading.."
+            let viewParsed (proj: Project) =
+                [ "<b>Status:</b> parsed correctly"
+                  sprintf "Project: %s" proj.Project ]
+                |> String.concat "<br />"
+            let viewFailed path error =
+                [ "<b>Status:</b> failed to load"; ""
+                  "<b>Error:</b>"
+                  error ]
+                |> String.concat "<br />"
+
+            { new TextDocumentContentProvider with
+                  member this.provideTextDocumentContent (uri: Uri) =
+                      match uri.path with
+                      | "/projects/status" ->
```

Comment:
yes, that `TextDocumentContentProvider` is something registered and internally used by ionide, so we create both the request urls and the handler

```fsharp
let projectStatusUri projectPath = vscode.Uri.parse(sprintf "fsharp-workspace://authority/projects/status?path=%s" (JS.encodeURIComponent(projectPath)))
```